### PR TITLE
CFE-4342: Added warning log message when OS is not recognized

### PR DESCRIPTION
**Pull request by olehermanse**
Merge together:

https://github.com/cfengine/core/pull/5460
https://github.com/cfengine/masterfiles/pull/2850

### CODE REVIEW

**Comment by olehermanse** [comment]
@cf-bottom Jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10512/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10512/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10512/

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/10513/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/10513/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-10513/

**Review by larsewi — APPROVED** [review]

**Review by nickanderson — APPROVED** [review]

**Review by craigcomstock — APPROVED** [review]

File: libenv/sysinfo.c [hunk 1]
```diff
@@ -3487,6 +3487,8 @@ static void SysOSNameHuman(EvalContext *ctx)
         EvalContextVariablePutSpecial(ctx, SPECIAL_SCOPE_SYS, lval,
                                       "Unknown", CF_DATA_TYPE_STRING,
                                       "source=agent");
+        Log(LOG_LEVEL_WARNING,
+            "Operating System not properly recognized, setting sys.os_name_human to \"Unkown\", please submit a bug report for us to fix this");
```
[the remark this answers is not in the document]
Comment (olehermanse) [reply libenv/sysinfo.c:3491]: @vpodzime Thanks, will fix 👀

[the remark this answers is not in the document]
Comment (olehermanse) [reply libenv/sysinfo.c:3491]: https://github.com/cfengine/core/pull/5463

**Comment by olehermanse** [comment]
Cherry-pick: https://github.com/cfengine/core/pull/5519

**Pull request merged**